components/script/dom/vrdisplay.rs

```diff
@@ -561,13 +605,28 @@ impl VRDisplay {
         *self.frame_data_receiver.borrow_mut() = Some(sync_receiver);
 
         let display_id = self.display.borrow().display_id;
-        let api_sender = self.layer_ctx.get().unwrap().webgl_sender();
+        let api_sender = match self.layer_ctx.get() {
+            Some(layer_ctx) => layer_ctx.webgl_sender(),
+            None => return,
+        };
         let js_sender = self.global().script_chan();
         let address = Trusted::new(&*self);
         let near_init = self.depth_near.get();
         let far_init = self.depth_far.get();
         let pipeline_id = self.global().pipeline_id();
 
+        // For displays with no external display, we use the regular window
+        // rAF, so we don't need to spin up a dedicated thread.
```

Comment:
How hard would it be to make the TestDisplay sync_poses behave in a way that we would not need to have this specific code path? I'm asking because the test will cover less code by re-using the window's rAF.

Reply:
IRC conversation: https://mozilla.logbot.info/servo/20190208#c15940619